core/client/app/controllers/settings/general.js

```diff
@@ -180,39 +190,53 @@ export default Controller.extend(SettingsSaveMixin, {
-                if (newUrl.match(/(?:twitter\.com\/)(\S+)/) || (!validator.isURL(newUrl) && newUrl.match(/([a-zA-Z0-9\.]+)/))) {
-                    let [ , username] = newUrl.match(/(?:twitter\.com\/)(\S+)/) || newUrl.match(/([a-zA-Z0-9\.]+)/);
-                    newUrl = `https://twitter.com/${username}`;
+            if (newUrl.match(/(?:twitter\.com\/)(\S+)/) || newUrl.match(/([a-z\d\.]+)/i)) {
+                let username = [];
 
-                    this.set('model.twitter', newUrl);
+                if (newUrl.match(/(?:twitter\.com\/)(\S+)/)) {
+                    [ , username] = newUrl.match(/(?:twitter\.com\/)(\S+)/);
+                } else {
+                    [username] = newUrl.match(/([^/]+)\/?$/gmi);
+                }
 
-                    this.get('model.errors').remove('twitter');
-                    this.get('model.hasValidated').pushObject('twitter');
+                // check if username starts with http or www and show error if so
+                if (username.match(/^(http|www)|(\/)/) || !username.match(/^[a-z\d\.\_]{1,15}$/gmi)) {
+                    errMessage = !username.match(/^[a-z\d\.\_]{1,15}$/gmi) ? 'Your Username is not a valid Twitter Username' : 'The URL must be in a format like https://twitter.com/yourUsername';
 
-                    // User input is validated
-                    return this.save().then(() => {
-                        this.set('model.twitter', '');
-                        run.schedule('afterRender', this, function () {
-                            this.set('model.twitter', newUrl);
-                        });
-                    });
-                } else if (validator.isURL(newUrl)) {
-                    errMessage = 'The URL must be in a format like ' +
-                        'https://twitter.com/yourUsername';
-                    this.get('model.errors').add('twitter', errMessage);
-                    this.get('model.hasValidated').pushObject('twitter');
-                    return;
-                } else {
-                    errMessage = 'The URL must be in a format like ' +
-                        'https://twitter.com/yourUsername';
                     this.get('model.errors').add('twitter', errMessage);
                     this.get('model.hasValidated').pushObject('twitter');
                     return;
                 }
+
+                newUrl = `https://twitter.com/${username}`;
+                this.set('model.twitter', newUrl);
+
+                this.get('model.errors').remove('twitter');
+                this.get('model.hasValidated').pushObject('twitter');
+
+                // User input is validated
+                return this.save().then(() => {
+                    this.set('model.twitter', '');
+                    run.schedule('afterRender', this, function () {
+                        this.set('model.twitter', newUrl);
+                    });
+                });
+            } else if (validator.isURL(newUrl)) {
```

Comment:
This `else if` block can be removed as it does exactly the same thing as the `else` block below.